### PR TITLE
Implement str.rpartition() based on CPython

### PR DESCRIPTION
**Pull request by densmirn**

### CODE REVIEW

**Comment by densmirn** [comment]
Could you please add the PR to the review queue?

**Comment by stuartarchibald** [comment]
> Could you please add the PR to the review queue?

done, thanks for the PR.

File: numba/unicode.py [hunk 1]
```diff
@@ -646,6 +646,42 @@ def count_impl(src, sub, start=start, end=end):
...
+
+    accepted = (types.UnicodeType, types.UnicodeCharSeq)
+    if thety is not None and not isinstance(thety, accepted):
+        msg = '"{}" must be {}, not {}'.format('sep', accepted, sep)
+        raise TypingError(msg)
+
+    def impl(data, sep):
+        # https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/stringlib/partition.h#L62-L115    # noqa: E501
+        empty_str = _empty_string(PY_UNICODE_4BYTE_KIND, 0, data._is_ascii)
+        sep_length = len(sep)
```
Comment (stuartarchibald) [inline numba/unicode.py]: Is it possible to optimise based on string kind here? If the string is ASCII does a 4byte kind need returning?

Comment (densmirn) [reply numba/unicode.py]: Changed. @stuartarchibald could you please review the change and if it's ok then move the PR on?

**Comment by densmirn** [comment]
Could you please queue the PR for review?

**Pull request merged**